src/parseTools.js

```diff
@@ -227,9 +227,8 @@ function splitI64(value) {
   const high = makeInlineCalculation(
       asmCoercion('Math.abs(VALUE)', 'double') + ' >= ' + asmEnsureFloat('1', 'double') + ' ? ' +
         '(VALUE > ' + asmEnsureFloat('0', 'double') + ' ? ' +
-        asmCoercion('Math.min(' + asmCoercion('Math.floor((VALUE)/' +
-        asmEnsureFloat(4294967296, 'double') + ')', 'double') + ', ' +
-        asmEnsureFloat(4294967295, 'double') + ')', 'i32') + '>>>0' +
+        asmCoercion('Math.floor((VALUE)/' +
+        asmEnsureFloat(4294967296, 'double') + ')', 'double') + '>>>0' +
```

Comment:
Why is it correct to remove the `Math.min` here?

I think the old code was correct. We can do rounding for the lower bits because we've ensured they are in the right range. But we need to use saturation because the value might be outside of range - imagine that the value is `2^70` plus some small value, then doing `>>>0` will do an odd wrapping operation (which can make it "randomly" lower or higher).

Reply:
The old code did saturation of the upper half produced this odd result:

```
0x1122334455667788AA -> 0xffffffff66780000
```

The new code truncates and produces the same result as BigInt64Array:

```
0x1122334455667788AA -> 0x2233445566780000
```

Reply:
"imagine that the value is 2^70 plus some small value" .. I'm not sure I understand doesn't the `Math.floor(($d)/ 4294967296.0)` take case of removing all the low bits?  Basically the same as `>> 32` in C/C++?

Reply:
We run this test both with and without WASM_BIGINT:

```
  @also_with_wasm_bigint                                                                             
  def test_parseTools(self):                                                                            
    self.do_other_test('test_parseTools.c', emcc_args=['--js-library', test_file('other/test_parseTools.js')])
```

The change in the expected output was initially generated because I changed WASM_BIGINT to always use `BigInt64Array`... then I updated splitI64 such that its output produces the same result.

Reply:
I belive that we already truncate int8, int16 and int32 beause we use `HEAP8[X] = Y`, etc.   If `Y` is larger than UINT8_MAX here I believe it gets truncated.  So this change just makes int64 consistent with all the other types.

It unlikely that anything good can happen if we try to write a number larger than INT64_MAX.. perhaps we should add an assertion in makeSetValue than the incoming value fits within the range?

Reply:
For example:

```
$ node
Welcome to Node.js v19.0.0.
Type ".help" for more information.
> const uint8 = new Uint8Array(2);
undefined
> uint8[0] = 300
300
> uint8[0]
44
> 
```

Reply:
True. Maybe I'm confused as to were we use `splitI64` then. I agree that when we store a bigger value into a small number of bits then we must naturally truncate. But `splitI64`, I thought, is used to approximate a number, not prepare it for storing.

If it is used to prepare for storing, then as I asked before, why is it even used in WASM_BIGINT mode? In that mode we can store directly.

Or is my confusion that it isn't used in that mode, and you fixed it for non-WASM_BIGINT mode?

Reply:
I see. Thanks, that all makes sense then!